src/components/input/code/views/inputView.jsx

```diff
@@ -0,0 +1,61 @@
+var React = require('react');
+
+module.exports = React.createClass({
+
+  propTypes: {
+    label: React.PropTypes.string,
+    type: React.PropTypes.oneOf(['text', 'password', 'email']),
+    placeholder: React.PropTypes.string,
+    defaultValue: React.PropTypes.string,
+    name: React.PropTypes.string,
+    id: React.PropTypes.string,
+    error: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    readOnly: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    valid: React.PropTypes.bool,
+    validator: React.PropTypes.string,
+    errorMessage: React.PropTypes.string,
+    onChange: React.PropTypes.func
+  },
+
+  getInitialState: function() {
+    return {
+      value: this.props.defaultValue || '',
+      error: null,
+      valid: true
+    }
+  },
+
+  getDefaultProps: function() {
+    return {
+      type: 'text',
+      disabled: false,
+      readOnly: false,
+      required: false,
+      onChange: this.changeHandler
+    }
+  },
+
+  changeHandler: function(e) {
+
+    var is_valid = true;
+
+    if(typeof this.props.validator !== 'undefined'){
+      is_valid = this.props.validator.test(e.target.value);
+    }
+
+    if( !is_valid && typeof this.props.errorMessage !== 'undefined'){
```

Comment:
You can get rid of the `else` part of this if statement by setting `this.props.errorMessage` to `null` in `getDefaultProps`.